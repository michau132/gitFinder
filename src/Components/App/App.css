.container__header {
    background-color: #B2AD00;
    color: #7619FF;
    display: grid;
    grid-template-columns: 150px 1fr  ;

}
.loop-box {
    align-self: center;
    justify-self: center;
    order: -1;
    width: 100px;
    height: 100px;
    background-color: transparent;
    display: flex;
    justify-content: center;
}
.loop {
    width: 40px;
    height: 40px;
    border-radius: 50%;
    background: linear-gradient(135deg, rgba(0,128,179,1) 0%, rgba(0,128,179,1) 1%, rgba(137,204,195,1) 28%, rgba(4,126,175,1) 60%, rgba(4,126,175,1) 100%);
    border: 6px solid black;
    box-shadow: -3px 3px 18px 0 #3C00FF;
    position:relative;
    animation-name: round;
    animation-duration: 12s;
    animation-iteration-count: infinite;
}
.loop:after {
    content: '';
    position: absolute;
    bottom: -16px;
    right: -33px;
    height: 2px;
    width: 38px;
    background-color: black;
    border: 2px solid black;
    transform: rotate(45deg);
    border-radius: 30px;
    box-shadow: 3px 7px 19px 0 #3C00FF;
}
.loop:before {
    content: '';
    position: absolute;
    top: 4px;
    left: 4px;
    height: 30px;
    width: 30px;
    border-radius: 50%;
    border-left: 2px solid black;
}
@keyframes round {
    50%{transform: rotate(90deg)}
}
