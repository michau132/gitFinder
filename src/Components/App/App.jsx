import React from 'react';
import {connect} from "react-redux";
import Form from '../Form/Form.jsx';
import UserInfo from '../UserInfo/UserInfo.jsx';
import {Route} from 'react-router';
import './App.css';
class App extends React.Component {
    render() {
        if (this.props.user.length === 0){
            return (
                <header className='container__header'>
                    <Route path='/' component={Form} />
                    <div className='loop-box'>
                        <div className="loop"></div>
                    </div>
                </header>
            )
        } else {
            return (
                <section>
                    <header className='container__header'>
                        <Route path='/' component={Form} />
                        <div className='loop-box'>
                            <div className="loop"></div>
                        </div>
                    </header>
                    <Route path='/user' component={UserInfo}/>
                </section>
            )
        }
    }
}

const mapStateToProps = (state) => {
    return state;
};

export default connect(mapStateToProps)(App);


