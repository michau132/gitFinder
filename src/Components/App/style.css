* {
    box-sizing: border-box;
}

.container {
    max-height: 100vh;
    margin: 0 auto;
    max-width: 1250px;
    padding: 20px;
    overflow: hidden;
}