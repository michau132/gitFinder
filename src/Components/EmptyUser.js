import React from 'react';
import { Typography } from '@material-ui/core';

const EmptyUser = () => (
  <Typography variant="h5">
    Here will be a profile from Github. Enter above a Github login to see more.
  </Typography>
);

export default EmptyUser;
