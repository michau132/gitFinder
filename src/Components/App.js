import React from 'react';
import {
  HashRouter,
  Route,
  Switch,
} from 'react-router-dom';
import { configure } from 'mobx';
import { Paper, withStyles } from '@material-ui/core';
import styled, { createGlobalStyle } from 'styled-components';
import UserContainer from '../Containers/UserContainer';
import HeaderContainer from '../Containers/HeaderContainer';
import EmptyUser from './EmptyUser';
import Header from './Header';
import User from './User';
import NotFound from './NotFound';

configure({ enforceActions: 'observed' });

const styles = {
  paper: {
    backgroundColor: '#F5F5F5',
  },
};

const GlobalStyle = createGlobalStyle`
  html {
    box-sizing: border-box;
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }
  body {
    background-color: #CCCCCC;
  }
  a {
    text-decoration: none;
  }
`;

const PaperWithPadding = styled(Paper)`
  padding: 8px;
`;

const App = ({ classes }) => (
  <HashRouter>
    <PaperWithPadding classes={{ root: classes.paper }}>
      <GlobalStyle />
      <PaperWithPadding>
        <HeaderContainer
          render={props => (
            <Header {...props} />
          )}
        />
        <Switch>
          <Route exact path="/" component={EmptyUser} />
          <Route
            exact
            path="/:user"
            render={() => (
              <UserContainer
                render={props => (
                  <User {...props} />
                )}
              />
            )}
          />
          <Route component={NotFound} />
        </Switch>
      </PaperWithPadding>
    </PaperWithPadding>
  </HashRouter>

);

export default withStyles(styles)(App);
