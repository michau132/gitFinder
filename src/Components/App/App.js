import React from 'react';
import {
  Route,
  Switch,
} from 'react-router-dom';
import UserContainer from '../../Containers/UserContainer';
import styles from './style.css';
import HeaderContainer from '../../Containers/HeaderContainer';
import EmptyUser from '../EmptyUser';

const App = () => (
  <section className={styles.container}>
    <HeaderContainer />
    <Switch>
      <Route exact path="/" component={EmptyUser} />
      <Route path="/:user" component={UserContainer} />
    </Switch>
  </section>
);

export default App;
