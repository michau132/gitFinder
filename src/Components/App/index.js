import React from 'react';
import {
  HashRouter,
  Route,
  Switch,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import UserContainer from '../../Containers/UserContainer';
import styles from './style.css';
import HeaderContainer from '../../Containers/HeaderContainer';
import EmptyUser from '../EmptyUser';
import store from '../../utils/store';


const App = () => (
  <Provider store={store}>
    <HashRouter>
      <section className={styles.container}>
        <HeaderContainer />
        <Switch>
          <Route exact path="/" component={EmptyUser} />
          <Route path="/:user" component={UserContainer} />
        </Switch>
      </section>
    </HashRouter>
  </Provider>

);

export default App;
