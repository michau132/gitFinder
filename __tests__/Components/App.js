import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../../src/Components/App';
import EmptyUser from '../../src/Components/EmptyUser';

describe('App', () => {
  describe('component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<App />).dive();
    });

    test('renders as expected', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders without crashing', () => {
      expect(wrapper).toBeDefined();
    });

    test('HeaderContainer should be defined', () => {
      expect(wrapper.find('withRouter(HeaderContainer)')).toBeDefined();
    });

    test('routes "/" to EmptyUser', () => {
      expect(wrapper.find('Route[exact=true][path="/"]').first().prop('component')).toBe(EmptyUser);
    });

    test('routes "/:user" to UserContainer', () => {
      expect(wrapper.find('Route[exact=true][path="/:user"]').first().prop('render')).toBeDefined();
    });
  });
});
