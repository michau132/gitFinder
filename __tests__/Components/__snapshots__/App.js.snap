// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders as expected 1`] = `
<HashRouter>
  <Styled(WithStyles(Paper))
    classes={
      Object {
        "root": "App-paper-1",
      }
    }
  >
    <GlobalStyleComponent />
    <Styled(WithStyles(Paper))>
      <withRouter(HeaderContainer)
        render={[Function]}
      />
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          exact={true}
          path="/:user"
          render={[Function]}
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </Styled(WithStyles(Paper))>
  </Styled(WithStyles(Paper))>
</HashRouter>
`;
